src/PAModel/Schemas/PcfControls/PcfControl.cs

```diff
@@ -0,0 +1,164 @@
+            }
+            if (dynamicControlDefinition.ContainsKey(PropertiesKey))
+            {
+                powerAppControl.Properties = Utilities.JsonParse<IEnumerable<Property>>(dynamicControlDefinition[PropertiesKey]);
+            }
+            if (dynamicControlDefinition.ContainsKey(AuthConfigPropertiesKey))
+            {
+                var authConfigPropertiesGroup = new List<IDictionary<string, AuthConfigProperty>>();
+                using (var doc = JsonDocument.Parse(dynamicControlDefinition[AuthConfigPropertiesKey]))
+                {
+                    var je = doc.RootElement;
+                    foreach (var section in je.EnumerateArray())
+                    {
+                        var authConfigProperties = new Dictionary<string, AuthConfigProperty>();
+                        foreach (var authProperty in section.EnumerateObject())
+                        {
+                            authConfigProperties.Add(authProperty.Name, authProperty.Value.ToObject<AuthConfigProperty>());
+                        }
+                        authConfigPropertiesGroup.Add(authConfigProperties);
+                    }
+                }
+                powerAppControl.AuthConfigProperties = authConfigPropertiesGroup;
+            }
+            if (dynamicControlDefinition.ContainsKey(DataConnectorsKey))
+            {
+                powerAppControl.DataConnectors = Utilities.JsonParse<IEnumerable<DataConnectorMetadata>>(dynamicControlDefinition[DataConnectorsKey]);
+            }
+            if (dynamicControlDefinition.ContainsKey(SubscribedFunctionalitiesKey))
+            {
+                powerAppControl.SubscribedFunctionalities = Utilities.JsonParse<Dictionary<string, string>>(dynamicControlDefinition[SubscribedFunctionalitiesKey]);
+            }
+            if (dynamicControlDefinition.ContainsKey(IncludedPropertiesKey))
+            {
+                powerAppControl.IncludedProperties = Utilities.JsonParse<IEnumerable<Property>>(dynamicControlDefinition[IncludedPropertiesKey]);
+            }
+            if (dynamicControlDefinition.ContainsKey(EventsKey))
+            {
+                powerAppControl.Events = Utilities.JsonParse<IEnumerable<Event>>(dynamicControlDefinition[EventsKey]);
+            }
+            if (dynamicControlDefinition.ContainsKey(CommonEventsKey))
+            {
+                powerAppControl.CommonEvents = Utilities.JsonParse<IEnumerable<Event>>(dynamicControlDefinition[CommonEventsKey]);
+            }
+            if (dynamicControlDefinition.ContainsKey(PropertyDependenciesKey))
+            {
+                powerAppControl.PropertyDependencies = Utilities.JsonParse<IEnumerable<PropertyDependency>>(dynamicControlDefinition[PropertyDependenciesKey]);
+            }
+
+            return powerAppControl;
+        }
+
+        internal static string GenerateDynamicControlDefinition(PcfControl control)
```

Comment:
same here, this could probably benefit from an intermediate poco representing the 2x encode step.